{{ cookiecutter.package_name }}/.github/workflows/ci.yml

```diff
@@ -90,21 +90,17 @@ jobs:
     secrets:
       CODECOV_TOKEN: {{ '${{ secrets.CODECOV_TOKEN }}' }}
 
-{% if cookiecutter.use_compiled_extensions == 'y' -%}
   publish:
-    # Build wheels when pushing to any branch except main
-    # publish.yml will only publish if tagged ^v.*
+    needs: [test, docs]
+    # Build wheels on PRs only when labelled. Releases will only be published if tagged ^v.*
+    # see https://github-actions-workflows.openastronomy.org/en/latest/publish.html#upload-to-pypi
     if: |
+      github.event_name != 'pull_request' ||
```

Comment:
I think it's probably overkill to skip wheel builds on main? For projects which do releases off main it's actively harmful, and for projects which use branches it's not a big deal?

Reply:
I have no issue with building wheels on branches or main. I think its better than finding out during a release.